src/ocm/components/clusterConfiguration/manifestsConfiguration/components/CustomManifestForm.tsx

```diff
@@ -78,24 +78,15 @@ export const CustomManifestForm = <CustomManifestFormValues extends FormViewMani
             (manifest) => manifest.fakeId === '',
           );
           await Promise.all(
-            manifestsThatExists.map(async (updatedManifest) => {
-              const existingFolderByFakeId = updatedManifest.fakeId?.substring(
-                0,
-                updatedManifest.fakeId.indexOf('#'),
-              );
-              const existingFolder: FormViewManifestFolder =
-                existingFolderByFakeId === 'manifests' || existingFolderByFakeId === 'openshift'
-                  ? existingFolderByFakeId
-                  : 'manifests';
-              const existingFileName = updatedManifest.fakeId?.substring(
-                updatedManifest.fakeId.indexOf('#') + 1,
-              );
+            manifestsThatExists.map((updatedManifest) => {
+              const { folderName, fileName } = getManifestDetails(updatedManifest);
+
               const existingManifest: FormViewManifest = {
-                folder: existingFolder,
-                filename: existingFileName || '',
+                folder: folderName,
+                filename: fileName || '',
                 manifestYaml: '',
               };
-              await ClustersService.updateCustomManifest(
+              void ClustersService.updateCustomManifest(
```

Comment:
I don't think this is correct either.
Since you are using `map` but are not actually returning anything, I can't see that `Promise.all` is receiving any array of promises. So even if it's not obvious looking at the code, I think each promise is still being sent during each iteration.

The pattern should be something like:

```

const individualPromises = manifestsThatExists.map(manifest =>ClustersService.updateCustomManifest(etc)) // No requests have been sent at this point

await Promise.all(individualPromises)  // requests are sent now
```